libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateRangePicker.tsx

```diff
@@ -42,66 +107,90 @@ class DateRangePickerComponent extends React.Component<IDateRangePickerProps & W
+                <DayPicker
+                    {...dayPickerPropsWithDefaults}
+                    onSelect={this.handleRangeSelect}
+                    selected={this.state.selectedRange}
+                    month={this.state.monthDate}
+                    classNames={classNameProps}
+                    onMonthChange={this.handleMonthChanged}
+                />
+            </div>
+        );
+
         const FromField = (
-            <DateTimePicker
+            <DateTimePickerWithInt
+                onKeyDown={this.onKeyDown}
+                ref={this.dateRangePickerInputFrom}
                 placeholderDate={intl.formatMessage({ id: "filters.from" })}
                 onChange={this.handleFromChange}
-                value={from}
+                value={this.state.inputFromValue}
                 dateFormat={dateFormat}
                 isMobile={isMobile}
-                locale={translationUtils.sanitizeLocaleForMoment(intl.locale)}
-                dayPickerPropsWithDefaults={dayPickerPropsWithDefaults}
                 handleDayClick={this.handleFromDayClick}
-                ref={this.fromInputRef}
                 isTimeEnabled={isTimeEnabled}
                 className={cx("s-date-range-picker-from", "gd-date-range-picker-from")}
                 error={typeof errorFrom !== "undefined"}
             />
         );
 
         const ToField = (
-            <DateTimePicker
+            <DateTimePickerWithInt
+                onKeyDown={this.onKeyDown}
+                ref={this.dateRangePickerInputTo}
                 placeholderDate={intl.formatMessage({ id: "filters.to" })}
                 onChange={this.handleToChange}
-                value={to}
+                value={this.state.inputToValue}
                 dateFormat={dateFormat}
                 isMobile={isMobile}
-                locale={translationUtils.sanitizeLocaleForMoment(intl.locale)}
-                dayPickerPropsWithDefaults={dayPickerPropsWithDefaults}
                 handleDayClick={this.handleToDayClick}
-                ref={this.toInputRef}
                 isTimeEnabled={isTimeEnabled}
                 className={cx("s-date-range-picker-to", "gd-date-range-picker-to")}
                 defaultTime={DAY_END_TIME}
                 error={typeof errorTo !== "undefined"}
             />
         );
 
+        const isFromInputDatePickerOpen = this.state.selectedInput === "from" && this.state.isOpen;
+        const isToInputDatePickerOpen = this.state.selectedInput === "to" && this.state.isOpen;
         return (
             <>
                 {isTimeEnabled ? (
-                    <div className="gd-date-range-picker datetime s-date-range-picker">
-                        <label>{intl.formatMessage({ id: "filters.from" })}</label>
-                        {FromField}
-                        <label>{intl.formatMessage({ id: "filters.to" })}</label>
-                        {ToField}
-                    </div>
+                    <>
```

Comment:
It seems to me that we would not need `<>` for this particular case. We would perhaps also do not need them in the second part of the condition, if 
```
{this.state.isOpen && DatePicker}
```
would be placed directly below {ToField} like in this part. 

Just curious, is there some reason for having it differently in each part of the condition?

Reply:
For time enabled, need to see which RangePicker is selected, so is attached to the correct input.